contrib/win32/win32compat/w32log.c

```diff
@@ -51,28 +51,25 @@ openlog(char *ident, unsigned int option, int facility)
 	if (logfd != -1 || ident == NULL)
 		return;
 
-	wchar_t path[PATH_MAX], log_file[PATH_MAX + 12];
+	wchar_t path[PATH_MAX] = { 0 }, log_file[PATH_MAX + 12] = { 0 };
 	if (GetModuleFileNameW(NULL, path, PATH_MAX) == 0)
 		return;
 
-	path[PATH_MAX - 1] = '\0';
+	path[PATH_MAX - 1] = L'\0';	
 
 	if (wcsnlen(path, MAX_PATH) > MAX_PATH - wcslen(logs_dir) )
 		return;
 
 	/* split path root and module */
 	{
-		wchar_t* tail = path + wcslen(path), *p;
+		wchar_t* tail = path + wcsnlen(path, MAX_PATH);
 		while (tail > path && *tail != L'\\' && *tail != L'/')
 			tail--;
 
-		memcpy(log_file, path, (tail - path) * sizeof(wchar_t));
-		p = log_file + (tail - path);
-		memcpy(p, logs_dir, wcslen(logs_dir) * sizeof(wchar_t));
-		p += 6;
-		memcpy(p, tail + 1, (wcslen(tail + 1) - 3) * sizeof(wchar_t));
-		p += wcslen(tail + 1) - 3;
-		memcpy(p, L"log\0", 8);
+		wcsncat_s(log_file, PATH_MAX + 12, path, tail - path);
+		wcsncat_s(log_file, PATH_MAX + 12, logs_dir, 6);
+		wcsncat_s(log_file, PATH_MAX + 12, tail + 1, wcslen(tail + 1) - 3);
+		wcsncat_s(log_file, PATH_MAX + 12, L"log", 3);
```

Comment:
Double check we are generating the logs right. I believe we don't have any automated test cases to verify this.